### ISSUE TITLE
[Very Important Feature Request] Add list of existing branches to choose in the Clone dialog.

### ISSUE THREAD

**Issue by XtroTheArctic** [ForkIssues/TrackerWin#948]
While working in big teams, the big projects contain many different branches greatly differentiating one from another.

A developer who needs to work on a specific branch must to be able to clone the repo using that branch only.

Cloning the master branch first and then switching to another branch is not ideal when the branches differentiate greatly, especially they contain different submodule links.

Most of the major git clients have this simple but powerful feature. Can you please add it?

**Comment by JeReT** [comment]
If you are using Git Flow you can select the starting branch for a feature / release / hotfix branch...
![image](https://user-images.githubusercontent.com/530561/93884003-9c1a5000-fce2-11ea-94af-84bc609ff1a1.png)

Strange that this option doesn't exist for plain branch creation...

PS: for git flow there is a UI bug: The hint text always shows "based on 'develop'", even if the branch is being changed.

**Comment by XtroTheArctic** [comment]
I am talking about Cloning a repo to the local computer for the first time; not creating a branch on an existing local repo.

**Comment by felixhirt** [comment]
i think this would be neat. But if you enter an url you dont have credentials for, the credential helper will pop up which can be confusing. Maybe check first if there are credentials for this host url and only try it if so?
edit: assuming the implementation would use git ls-remote

**Comment by ChefkeGremmen** [comment]
yes, I want this as well! switch between branches after cloning is cumbersome (imho).

**Comment by PlasmaHH** [comment]
This is one of the few things I am missing, having used it in sourcetree ( additionally the option to do recurse subtree or not in the dialogue)
